<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>TaskPanelCfdDynamicMesh</class>
 <widget class="QWidget" name="TaskPanelCfdDynamicMesh">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>436</width>
    <height>1078</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Dynamic Mesh</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="3" column="1">
    <spacer name="verticalSpacer_3">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="0" column="1">
    <widget class="QComboBox" name="cb_dynamic_type"/>
   </item>
   <item row="0" column="0">
    <widget class="QLabel" name="label_10">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>Dynamic mesh type</string>
     </property>
    </widget>
   </item>
   <item row="2" column="0" colspan="2">
    <widget class="QStackedWidget" name="stackedWidget">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="page">
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>9</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>9</number>
       </property>
       <item>
        <widget class="QFrame" name="triggerFieldFrame">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="frameShape">
          <enum>QFrame::NoFrame</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Raised</enum>
         </property>
         <layout class="QGridLayout" name="extrusionLayout">
          <item row="3" column="1">
           <widget class="QLabel" name="label_5">
            <property name="text">
             <string>Refine when above</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLabel" name="label_6">
            <property name="font">
             <font>
              <weight>75</weight>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>Trigger field</string>
            </property>
           </widget>
          </item>
          <item row="3" column="2" colspan="2">
           <widget class="Gui::InputField" name="if_lower_refinement">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="toolTip">
             <string extracomment="Select 0 to use default value">&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Where the refinement field lies between these values, the mesh will continue to be refined until the max refinement level is reached&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>0.0</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
            <property name="singleStep" stdset="0">
             <double>1.000000000000000</double>
            </property>
            <property name="maximum" stdset="0">
             <double>1000000000.000000000000000</double>
            </property>
            <property name="unit" stdset="0">
             <string notr="true"/>
            </property>
            <property name="format" stdset="0">
             <string>d</string>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>Refine when below</string>
            </property>
           </widget>
          </item>
          <item row="4" column="2" colspan="2">
           <widget class="Gui::InputField" name="if_upper_refinement">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="toolTip">
             <string extracomment="Select 0 to use default value">&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Field variable level above which the mesh is refined&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>0.0</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
            <property name="singleStep" stdset="0">
             <double>1.000000000000000</double>
            </property>
            <property name="maximum" stdset="0">
             <double>1000000000.000000000000000</double>
            </property>
            <property name="unit" stdset="0">
             <string notr="true"/>
            </property>
            <property name="format" stdset="0">
             <string>d</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLabel" name="label_7">
            <property name="text">
             <string>Refinement field</string>
            </property>
           </widget>
          </item>
          <item row="6" column="1">
           <widget class="QLabel" name="label_8">
            <property name="text">
             <string>Unrefine when below</string>
            </property>
           </widget>
          </item>
          <item row="6" column="2" colspan="2">
           <widget class="Gui::InputField" name="if_unrefine_level">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="toolTip">
             <string extracomment="Select 0 to use default value">&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Where the refinement field is below this value and not marked for refinement, the mesh will be repeatedly unrefined until the original mesh is recovered&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>0.0</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
            <property name="singleStep" stdset="0">
             <double>1.000000000000000</double>
            </property>
            <property name="maximum" stdset="0">
             <double>1000000000.000000000000000</double>
            </property>
            <property name="unit" stdset="0">
             <string notr="true"/>
            </property>
            <property name="format" stdset="0">
             <string>d</string>
            </property>
           </widget>
          </item>
          <item row="1" column="2" colspan="2">
           <widget class="QLineEdit" name="le_refinement_field">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Name of a scalar field to control refinement&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QFrame" name="dynamicMeshFrame">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <layout class="QGridLayout" name="gridLayout_11">
          <property name="leftMargin">
           <number>6</number>
          </property>
          <property name="topMargin">
           <number>6</number>
          </property>
          <property name="rightMargin">
           <number>6</number>
          </property>
          <property name="bottomMargin">
           <number>6</number>
          </property>
          <item row="8" column="1">
           <widget class="QLabel" name="label_9">
            <property name="text">
             <string>Max cells</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>Max refinement level</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QLabel" name="label">
            <property name="text">
             <string>Buffer layers</string>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QLabel" name="label_1">
            <property name="text">
             <string>Refinement interval</string>
            </property>
           </widget>
          </item>
          <item row="8" column="3">
           <widget class="Gui::InputField" name="if_max_cells">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Limit imposed on total number of cells in the mesh&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="singleStep" stdset="0">
             <double>1.000000000000000</double>
            </property>
            <property name="unit" stdset="0">
             <string notr="true"/>
            </property>
            <property name="format" stdset="0">
             <string>d</string>
            </property>
           </widget>
          </item>
          <item row="1" column="3">
           <widget class="QDoubleSpinBox" name="sb_max_refinement_levels">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Maximum number of levels of refinement to apply&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="decimals">
             <number>0</number>
            </property>
            <property name="minimum">
             <double>1.000000000000000</double>
            </property>
            <property name="maximum">
             <double>99.000000000000000</double>
            </property>
            <property name="singleStep">
             <double>1.000000000000000</double>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLabel" name="label_4">
            <property name="font">
             <font>
              <weight>75</weight>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>Refinement controls</string>
            </property>
           </widget>
          </item>
          <item row="4" column="3">
           <widget class="QDoubleSpinBox" name="sb_refinement_interval">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Iteration interval at which to perform dynamic mesh refinement&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="decimals">
             <number>0</number>
            </property>
            <property name="minimum">
             <double>0.000000000000000</double>
            </property>
            <property name="maximum">
             <double>999.000000000000000</double>
            </property>
            <property name="singleStep">
             <double>1.000000000000000</double>
            </property>
            <property name="value">
             <double>1.000000000000000</double>
            </property>
           </widget>
          </item>
          <item row="3" column="3">
           <widget class="QDoubleSpinBox" name="sb_no_buffer_layers">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Number of cells layers between refinement and existing cells&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="decimals">
             <number>0</number>
            </property>
            <property name="minimum">
             <double>1.000000000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QFrame" name="surfaceOrInernalVolume">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="frameShape">
          <enum>QFrame::NoFrame</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Raised</enum>
         </property>
         <layout class="QGridLayout" name="gridLayout_3">
          <item row="0" column="0">
           <widget class="QCheckBox" name="cb_write_refinement_volscalarfield">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Write refinement cell level scalar at each solver write iteration&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>Output refinement level field</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item row="1" column="0" colspan="2">
    <widget class="QLabel" name="l_description">
     <property name="text">
      <string>TextLabel</string>
     </property>
     <property name="wordWrap">
      <bool>true</bool>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Gui::InputField</class>
   <extends>QLineEdit</extends>
   <header>Gui/InputField.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>cb_dynamic_type</tabstop>
  <tabstop>sb_refinement_interval</tabstop>
  <tabstop>if_max_cells</tabstop>
  <tabstop>le_refinement_field</tabstop>
  <tabstop>if_lower_refinement</tabstop>
  <tabstop>if_upper_refinement</tabstop>
  <tabstop>if_unrefine_level</tabstop>
  <tabstop>cb_write_refinement_volscalarfield</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
